src/python/zquantum/core/circuits/symbolic/qiskit_expressions.py

```diff
@@ -36,8 +36,8 @@ def _expr_from_qiskit_param_expr(
     # At the moment of writing this the qiskit version that we use (0.23.2) as well
     # as the newest version 0.23.5) does not provide a better way to access symbolic
     # expression wrapped by ParameterExpression.
-    sympy_expr = qiskit_expr._symbol_expr
-    return expression_from_sympy(sympy_expr)
+    symengine_expr = qiskit_expr._symbol_expr
+    return expression_from_symengine(symengine_expr)
```

Comment:
Does changing this mean that from now on we'll be compatible with only the newest Qiskit version (`0.28`?) and not with the previous ones? If that's the case, WDYT about adding backwards compatibility with older Qiskit versions that use `sympy` instead of `symengine`?

Reply:
Good catch. At some point, it would be great to merge sympy/symengine translations into one coherent set of functions, which would make `core` compatible with current and older versions of qiskit. However, for now, I suggest a simple route in which we just try both of the translations and see if one worked. It is implemented in 7174e33, WDYT?

Reply:
Makes sense!